### ISSUE TITLE
MS_loss is calculated on reduction.before_neck instead of reduction.after_neck

### ISSUE THREAD

**Issue by KenAdamJoseph** [jixunbo/LightMBN#29]
before_neck and after_neck are two variables in reduction blocks. (after_neck = self.bn(before_neck))

During testing, after_neck is used.
However, during training, MS_loss is calculated on before_neck, instead of after_neck.

Experimental results show it is the better choice, but is there any intuitive explanation?

Best

**Comment by fubel** [comment]
Hey,

have a look at _Figures 5_ and _6_ of the [Bag of Tricks paper](https://arxiv.org/pdf/1903.07071.pdf). The idea is to apply metric losses (such as triplet loss or ms loss) before the neck, the ID loss (i.e., cross entropy) after the FC layer, and for inference, we use the features before the FC layer but after the neck. 

Fabian

**Comment by KenAdamJoseph** [comment]
The paper explains it well.  Thanks.
